### PR TITLE
[Maker Cleanup] Remove deprecated maker app pathway code (Part 3)

### PR DESCRIPTION
**Pull request by fisher-alice**
Now that WebSerial has been implemented for both the Circuit Playground and micro:bit since early spring 2023, this PR removes references to the Maker App pathway. This is part 3 of a series of 3 PRs - [Part 1](https://github.com/code-dot-org/code-dot-org/pull/52399) removes code related to the chrome app, [Part 2](https://github.com/code-dot-org/code-dot-org/pull/52426) removes code related to 'maker-captouch' experiment.

Notification banner BEFORE UPDATE in Maker App at 'maker/setup':

Screen Shot 2023-06-26 at 5 27 26 PM

Notification banner AFTER UPDATE in Maker App at 'maker/setup':
Note that the wording has slightly changed and the color is now red instead of yellow. @moneppo - lmk if this is okay - thanks!

Screen Shot 2023-06-24 at 3 36 55 PM

We no longer use Node `SerialPort` and use only `WebSerial` to connect to the board, and I was able to remove code related to Downloads. 🎉 

Note that I added a re-added a unit test in `SetupChecklistTest.js` that checks that the page reloads on 're-detect' on successful. I refactored the `redetect` function to always reload in `SetupChecklist` in [this prior PR](https://github.com/code-dot-org/code-dot-org/pull/52399) which includes the rationale. 

## Links
[jira ticket](https://codedotorg.atlassian.net/browse/SL-659) 

## Testing story
I created a [draft PR](https://github.com/code-dot-org/code-dot-org/pull/52443) that compared this branch (includes changes from all 3 Maker cleanup PRs) with staging to kick off a drone build and the build was successful. I checked locally on the browser and the Maker App to ensure I can still connect successfully with both the micro:bit and CP and run programs as expected.

## Follow-up work
Support articles must be updated to correspond with deprecation. @moneppo 

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/lib/kits/maker/ui/SetupGuide.jsx [hunk 1]
```diff
@@ -69,10 +69,20 @@ export default class SetupGuide extends React.Component {
...
+          <Notification
+            type={NotificationType.failure}
+            notice={i18n.makerAppDeprecationNoticeTitle()}
+            details={i18n.makerAppDeprecationNoticeDetails()}
+            detailsLinkText={i18n.makerDeprecationNoticeLinkText()}
+            detailsLink={MAKER_DEPRECATION_SUPPORT_URL}
+            dismissible
+          />
+        )}
+        {!isCodeOrgBrowser() && chromeVersion && chromeVersion < 90 && (
```
Comment (sanchitmalhotra126) [inline apps/src/lib/kits/maker/ui/SetupGuide.jsx]: reuse constant for chrome version?

File: apps/src/lib/kits/maker/ui/SetupInstructions.jsx [hunk 2]
```diff
@@ -129,332 +100,3 @@ class Support extends React.Component {
     );
   }
 }
-
-class Downloads extends React.Component {
```
Comment (sanchitmalhotra126) [inline apps/src/lib/kits/maker/ui/SetupInstructions.jsx:133]: 🧹 🎉

**Review by sanchitmalhotra126 — APPROVED** [review]
Feels great to see all this branching logic go away! Nice!

File: apps/src/lib/kits/maker/ui/SetupGuide.jsx (hunk 1 above)
Comment (fisher-alice) [reply apps/src/lib/kits/maker/ui/SetupGuide.jsx]: Thanks for catching this!

**Pull request merged**